fast64_internal/oot/scene/exporter/to_c/scene_cutscene.py

```diff
@@ -1,144 +1,95 @@
-            elif list.listType in ["PlayBGM", "StopBGM", "FadeBGM"]:
-                data.source += e.value
-                if list.listType != "FadeBGM":
-                    data.source += " + 1"  # Game subtracts 1 to get actual seq
-                data.source += ", " + str(e.startFrame) + ", " + str(e.endFrame) + ", 0, 0, 0, 0, 0, 0, 0, 0"
-            elif list.listType == "Misc":
-                data.source += (
-                    str(e.operation)
-                    + ", "
-                    + str(e.startFrame)
-                    + ", "
-                    + str(e.endFrame)
-                    + ", 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0"
-                )
-            elif list.listType == "0x09":
-                data.source += (
-                    "0, "
-                    + str(e.startFrame)
-                    + ", "
-                    + str(e.startFrame + 1)
-                    + ", "
-                    + e.unk2
-                    + ", "
-                    + e.unk3
-                    + ", "
-                    + e.unk4
-                    + ", 0, 0"
-                )
-            elif list.listType == "Unk":
-                data.source += (
-                    e.unk1
-                    + ", "
-                    + e.unk2
-                    + ", "
-                    + e.unk3
-                    + ", "
-                    + e.unk4
-                    + ", "
-                    + e.unk5
-                    + ", "
-                    + e.unk6
-                    + ", "
-                    + e.unk7
-                    + ", "
-                    + e.unk8
-                    + ", "
-                    + e.unk9
-                    + ", "
-                    + e.unk10
-                    + ", "
-                    + e.unk11
-                    + ", "
-                    + e.unk12
+                    csData.source += f"{e.startFrame}, {e.endFrame}"
+
+                elif e.textboxType == "OcarinaAction":
+                    csData.source += f"{e.ocarinaAction}, {e.startFrame}, {e.endFrame}, {e.ocarinaMessageId}"
+
+            elif list.listType == "LightSettingsList":
+                # the endFrame variable is not used in the implementation of the commend
+                # so the value doesn't matter
+                csData.source += f"{e.lightSettingsIndex}, {e.startFrame}" + (", 0" * 9)
+
+            elif list.listType == "TimeList":
+                # same as above
+                csData.source += f"0, {e.startFrame}, 0, {e.hour}, {e.minute}"
+
+            elif list.listType == "RumbleList":
+                # same as above
+                csData.source += (
+                    f"0, {e.startFrame}, 0, {e.rumbleSourceStrength}, {e.rumbleDuration}, {e.rumbleDecreaseRate}, 0, 0"
                 )
+
+            elif list.listType in ["StartSeqList", "StopSeqList", "FadeOutSeqList"]:
+                endFrame = e.endFrame if list.listType == "FadeOutSeqList" else "0"
```

Comment:
why do you force endFrame to be 0?

Reply:
in decomp it says the end frame isn't used to the value doesn't matter at all (unless I misread the line, which is possible)

Reply:
(https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/include/z64cutscene_commands.h#L197 & https://github.com/zeldaret/oot/blob/9f09505d34619883748a7dab05071883281c14fd/include/z64cutscene_commands.h#L212)

Reply:
but yea I guess I can use the default value of the class, those need a small cleanup anyway but idk if I should do it in this PR

Reply:
It should be noted that decomp's implementation of the cutscene structs doesn't match the original implementation. For example, all evidence suggests that `CsCmdMisc`, `CsCmdLightSetting`, `CsCmdStartSeq`, `CsCmdStopSeq` `CsCmdFadeOutSeq` are all the same type as `CsCmdActorCue`. Decomp just implements them separately so that they can hide editor only properties not read by the game, and give a few properties more specialized names.

`endFrame` is one such property that all commands have, but isn't always needed by the game. But I don't think it makes any sense to deliberately force the value to 0 just because you can get away with it.

Reply:
lol
obv mzx and I dont agree on this, shown by the discussions in the z_demo pr for this topic. but imo its fine to use 0 when it doesnt matter

Reply:
i understand your point of view a little from the perspective of vanilla data.

but for custom cutscenes, how is it corrupt data? you want the commands that dont use an endframe to have one anyway and be confusing for the user? the ui might as well hide it so people dont need to think about it. cause again, it does not matter

Reply:
So, with your (mzx) logic we should also use 1 for unused args? It doesn't make sense to me, I mean I understand the point of view, like ok I get it originally it wasn't split etc, but for Fast64 in a modding context where I and other devs try our best to get something easy to use for everyone I don't think consistency should be considered for this particular thing as it's just confusing, alternatively I can accept using a macro but idk it doesn't seem really useful, also I don't really understand how 0 is corrupted data, especially for a frame value where you can actually want 0 to be used (well, ok 0 for a end frame is weird but you get what I mean hopefully)

Reply:
Tldr I just want Fast64 to be easy to use and understand and having a value for this would be confusing for the users

Reply:
Two simple reasons:

* Storing the `endFrame` for all commands is a nice thing to have, and the structs already support this.
* Given that the structs all call the property `endFrame` in decomp, it makes zero sense to deliberately generate incorrect data for this property, since the user could have their own reasons for reading the data.